### PR TITLE
Add clear text icon to search fields

### LINKED ISSUE #425: Clearing-cross for search field
When searching an object via the search box on the rock and route list, a clearing cross should appear at the end of the search box to remove everything that has been typed into the box yet.

We should use the Andoid SearchView widget, which provides this function already: https://developer.android.com/reference/android/widget/SearchView

### PR DESCRIPTION
**Pull request by chrgernoe**
Fixes #425

### CODE REVIEW

File: app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt [hunk 1]
```diff
@@ -52,8 +53,10 @@ class SearchBarHandler(searchBarLayout: ConstraintLayout,
             })
         }
 
-        searchBarLayout.findViewById<EditText>(R.id.searchEditText).apply {
+        searchBarLayout.findViewById<TextInputLayout>(R.id.searchTextInputLayout).apply {
             setHint(searchHintResource)
+        }
+        searchBarLayout.findViewById<EditText>(R.id.searchEditText).apply {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt:59]: You have changed the widget's type in the xml to `com.google.android.material.textfield.TextInputEditText`. Does this still correspond to Kotlin's `EditText` type?

File: app/src/main/res/layout/activity_partners.xml [hunk 2]
```diff
@@ -22,26 +22,37 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
+    <com.google.android.material.textfield.TextInputLayout
```
Comment (paetz) [inline app/src/main/res/layout/activity_partners.xml:25]: Why did you move the TextField out of the `LinearLayout`?  Do we actually still need the `LinearLayout` below (It only contains one element in your solution)?

File: app/src/main/res/layout/searchbar.xml [hunk 3]
```diff
@@ -27,21 +28,32 @@
...
-        android:layout_height="45dp"
-        android:layout_marginTop="8dp"
-        android:maxLines="1"
-        android:singleLine="true"
-        app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+    <com.google.android.material.textfield.TextInputLayout
```
Comment (paetz) [inline app/src/main/res/layout/searchbar.xml:38]: Maybe it makes sense for `PartnersActivity` to include this searchbar and make the checkbox above invisible at the activity's start-up instead of defining the same component on its own? Doing so, we could reduce quite some code-duplication.

Comment (paetz) [reply app/src/main/res/layout/searchbar.xml:38]: I could even think of re-using the checkbox for PartnerActivity e.g. for sth. like "sort by name" to either sort partners by name or number of shared routes.

**Review by paetz — DISMISSED** [review]

File: app/src/main/res/layout/searchbar.xml (hunk 3 above)
Comment (chrgernoe) [reply app/src/main/res/layout/searchbar.xml:38]: I thought about this, too but I didn't want to change too many things. Anyway, I will include this change into the the PR.

File: app/src/main/res/layout/activity_partners.xml (hunk 2 above)
Comment (chrgernoe) [reply app/src/main/res/layout/activity_partners.xml:25]: Because I need the layout constraint feature for correct placement of the search field. The `LinearLayout` acts as container of the included XML so that the list view can be aligned correctly in the current layout. Since the included XML is used in different layouts, I probably break some other layout. Another PR is required, to unify all the layouts where the list view is included.

File: app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt (hunk 1 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt:59]: It's a sub-class of `EditText` (https://developer.android.com/reference/com/google/android/material/textfield/TextInputEditText) which shall be used within the `TextInputLayout`.

File: app/src/main/java/com/yacgroup/yacguide/PartnersActivity.kt [hunk 4]
```diff
@@ -75,6 +79,12 @@ class PartnersActivity : AppCompatActivity() {
             }
         })
 
+        _searchBarHandler = SearchBarHandler(
+            searchBarLayout = findViewById(R.id.searchBarLayout),
+            checkBoxVisibility = INVISIBLE,
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/PartnersActivity.kt:84]: Please use `View.INVISIBLE` and remove the `android.view.View.INVISIBLE` import

File: app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt [hunk 5]
```diff
@@ -22,38 +23,45 @@ import android.content.SharedPreferences
...
-                       checkBoxDefaultValue: Boolean,
-                       private val _settings: SharedPreferences,
-                       private val _initCallback: (Boolean) -> Unit,
-                       private val _updateCallback: (String, Boolean) -> Unit) {
+                       checkBoxTitle: String = "",
+                       checkBoxDefaultValue: Boolean = false,
+                       private val _settings: SharedPreferences? = null,
+                       private val _initCallback: ((Boolean) -> Unit)? = null,
+                       private val _updateCallback: ((String, Boolean) -> Unit)? = null,
+                       checkBoxVisibility: Int = VISIBLE) {
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt:41]: - [ ] Please use `View.INVISIBLE` and remove the `android.view.View.INVISIBLE` import
- [ ] Please put this parameter 3 lines above (behind checkBoxDefaultValue)

File: app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt [hunk 6]
```diff
@@ -22,38 +23,45 @@ import android.content.SharedPreferences
...
                        searchHintResource: Int,
-                       checkBoxTitle: String,
-                       checkBoxDefaultValue: Boolean,
-                       private val _settings: SharedPreferences,
-                       private val _initCallback: (Boolean) -> Unit,
-                       private val _updateCallback: (String, Boolean) -> Unit) {
+                       checkBoxTitle: String = "",
+                       checkBoxDefaultValue: Boolean = false,
+                       private val _settings: SharedPreferences? = null,
+                       private val _initCallback: ((Boolean) -> Unit)? = null,
```
Comment (paetz) [inline app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt:39]: Wouldn't it be safer (and more convenient) to default `_initCallback` and `_updateCallback` to empty lambdas?
--> `_initCallback: (Boolean) -> Unit = {}`

File: app/src/main/res/layout/activity_partners.xml [hunk 7]
```diff
@@ -25,23 +26,16 @@
     <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:orientation="vertical"
         android:clickable="true"
         android:focusable="true"
         android:focusableInTouchMode="true"
+        android:orientation="vertical"
+        app:layout_constraintEnd_toEndOf="parent"
```
Comment (paetz) [inline app/src/main/res/layout/activity_partners.xml:33]: Why is this necessary?

**Review by paetz — DISMISSED** [review]

File: app/src/main/res/layout/activity_partners.xml (hunk 7 above)
Comment (chrgernoe) [reply app/src/main/res/layout/activity_partners.xml:33]: You mean the `LinearLayout`? I answered the question already above?

File: app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt (hunk 5 above)
Comment (chrgernoe) [reply app/src/main/java/com/yacgroup/yacguide/utils/SearchBarHandler.kt:41]: Because the class is used without named arguments in the code, I wanted to make it compatible. But anyway, I will change those lines, too.

File: app/src/main/res/layout/activity_partners.xml (hunk 7 above)
Comment (paetz) [reply app/src/main/res/layout/activity_partners.xml:33]: I mean `app:layout_constraintEnd_toEndOf="parent"`

Comment (chrgernoe) [reply app/src/main/res/layout/activity_partners.xml:33]: You are right, this is not necessary. I going to remove it.
